tests/digraph/test_depth.py

```diff
@@ -69,3 +69,111 @@ def test_empty_graph(self):
         self.assertEqual([], retworkx.dag_longest_path(dag))
+
+    def test_parallel_edges(self):
+        dag = retworkx.PyDiGraph()
+        dag.extend_from_weighted_edge_list(
+            [
+                (0, 1, 1),
+                (0, 3, 1),
+                (3, 4, 1),
+                (4, 5, 1),
+                (1, 2, 1),
+                (0, 1, 3),
+            ]
+        )
+        self.assertEqual(
+            [0, 3, 4, 5],
+            retworkx.dag_longest_path(dag),
+        )
+
+    def test_linear_with_weight(self):
+        """Longest depth for a simple dag.
+
+        a
+        |
+        b
+        |\
+        c d
+        |\
+        e |
+        | |
+        f g
+        """
+        dag = retworkx.PyDAG()
+        node_a = dag.add_node("a")
+        node_b = dag.add_child(node_a, "b", 4)
+        node_c = dag.add_child(node_b, "c", 4)
+        dag.add_child(node_b, "d", 5)
+        node_e = dag.add_child(node_c, "e", 2)
+        dag.add_child(node_e, "f", 2)
+        node_g = dag.add_child(node_c, "g", 15)
+        self.assertEqual(
+            [node_a, node_b, node_c, node_g],
+            retworkx.dag_longest_path(dag, lambda _, __, weight: weight),
+        )
+        self.assertEqual(
+            23,
+            retworkx.dag_longest_path_length(dag, lambda _, __, weight: weight),
+        )
+
+    def test_parallel_edges_with_weights(self):
+        dag = retworkx.PyDiGraph()
+        dag.extend_from_weighted_edge_list(
+            [
+                (0, 1, 1),
+                (0, 3, 1),
+                (3, 4, 1),
+                (4, 5, 1),
+                (1, 2, 1),
+                (0, 1, 3),
+            ]
+        )
+        self.assertEqual(
+            [0, 1, 2],
+            retworkx.dag_longest_path(dag, lambda _, __, weight: weight),
+        )
+        self.assertEqual(
+            4,
+            retworkx.dag_longest_path_length(
+                dag, weight_fn=lambda _, __, weight: weight
+            ),
+        )
+
+    def test_less_linear_with_weight(self):
+        dag = retworkx.PyDAG()
+        node_a = dag.add_node("a")
+        node_b = dag.add_child(node_a, "b", 1)
+        node_c = dag.add_child(node_b, "c", 1)
+        node_d = dag.add_child(node_c, "d", 1)
+        node_e = dag.add_child(node_d, "e", 1)
+        dag.add_edge(node_a, node_c, 3)
+        dag.add_edge(node_a, node_e, 3)
+        dag.add_edge(node_c, node_e, 3)
+        self.assertEqual(
+            6,
+            retworkx.dag_longest_path_length(
+                dag, weight_fn=lambda _, __, weight: weight
+            ),
+        )
+        self.assertEqual(
+            [node_a, node_c, node_e],
+            retworkx.dag_longest_path(
+                dag, weight_fn=lambda _, __, weight: weight
+            ),
+        )
+
+    def test_degenerate_graph_with_weight(self):
+        dag = retworkx.PyDAG()
+        dag.add_node(0)
+        self.assertEqual([0], retworkx.dag_longest_path(dag, weight_fn=int))
```

Comment:
I'm a bit confused with `weight_fn=int` here. Isn't `weight_fn` supposed to take 3 arguments? I believe this would raise a `TypeError` if the graph was not trivial.

Reply:
Oops, yeah it's just sloppy copy and paste errors from me. You're correct this would raise a type error if the weight_fn was ever called. I'll fix this and add some negative tests to catch the type error case too.

Reply:
Done in https://github.com/Qiskit/retworkx/pull/383/commits/fb5f28ab687f789d8dab5841cdd66d84587ad581

Reply:
It seems that you fixed it for the function call below but not at this line.

Reply:
sigh, not sure how I missed that, one sec I'll update it for real

Reply:
ok, actually fixed in: https://github.com/Qiskit/retworkx/pull/383/commits/d5167ecc218d2c485a73c8ef288e170f124720e3